dlt-database/entity/0001-init_db/Account.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  Entity,
+  PrimaryGeneratedColumn,
+  Column,
+  CreateDateColumn,
+  ManyToOne,
+  JoinColumn,
+  OneToOne,
+  OneToMany,
+  ManyToMany,
+  JoinTable,
+} from 'typeorm'
+import { User } from './User'
+import { Community } from './Community'
+import { TransactionDraft } from './TransactionDraft'
+import { ConfirmedTransaction } from './ConfirmedTransaction'
+
+@Entity('accounts')
+export class Account {
+  @PrimaryGeneratedColumn('increment', { unsigned: true })
+  id: number
+
+  @ManyToOne(() => User, (user) => user.accounts) // Assuming you have a User entity with 'accounts' relation
+  @JoinColumn({ name: 'user_id' })
+  user: User
+
+  // if user id is null, account belongs to community gmw or auf
+  @Column({ name: 'user_id', type: 'int', unsigned: true, nullable: true })
+  userId?: number
+
+  @Column({ name: 'account_nr', type: 'int', unsigned: true, default: 0 })
+  accountNr: number
+
+  @Column({ type: 'binary', length: 32, unique: true })
+  pubkey: Buffer
+
+  @Column({ type: 'tinyint', unsigned: true })
+  type: number
+
+  @CreateDateColumn({
+    name: 'created_at',
+    type: 'datetime',
+    default: () => 'CURRENT_TIMESTAMP',
+    nullable: false,
+  })
+  createdAt: Date
+
+  @Column({ name: 'confirmed_at', type: 'datetime', nullable: true })
+  confirmedAt?: Date
+
+  @OneToOne(() => Community, (community) => community.gmwAccount)
+  gmwCommunity?: Community
+
+  @OneToOne(() => Community, (community) => community.aufAccount)
+  aufCommunity?: Community
+
+  @ManyToMany(() => Community, (community) => community.communityAccounts)
+  @JoinTable({
+    name: 'accounts_communities',
+    joinColumn: { name: 'account_id', referencedColumnName: 'id' },
+    inverseJoinColumn: { name: 'community_id', referencedColumnName: 'id' },
+  })
+  accountCommunities: Community[]
```

Comment:
This is for assigning when which accounts belong to which commuity. That means when an account moves, it gets a new entry. For this are the two date fields which describe the period: __valid_from__ and __valid_to__.

Reply:
Hmm i understand your scenario, but is it right to say that each community will have its own blockchain?
An account movement will lead to set the valid_to in the source-community's blockchain and to set the valid_from for the dest-community's blockchain with a new account?
Or do you need this construction for the overall blockchain of all communities? But then it is the question if the databasemodel should be splitted in two models for the different requirements?

Reply:
Every community has his own blockchain, that is right.
This db model is designed for keeping track of multiple chains, so it can check cross-group transactions.
The same as my Gradido Node Server.
An account movement lead to updating valid_to of the last entry and creating a new entry in account_communities for the same account setting valid_from to the same value as valid_to from the old entry.
It can be used for keeping track of all blockchains, but it isn't for the overall blockchain of all communities.

Reply:
Ok i see...
These are details about your blockchain requirements and concept in a detail i had not reached yet.
For sure i will get it step by step...